vaccine/states.py

```diff
@@ -254,3 +265,39 @@ async def display(self, message: Message):
                 ],
             },
         )
+
+
+class SectionedChoiceState(ChoiceState):
+    def __init__(
+        self,
+        *args,
+        sections: Sequence[Tuple[str, Sequence[Choice]]],
+        separator: Optional[str] = None,
+        **kwargs,
+    ):
+        choices: List[Choice] = []
+        for section_choices in sections:
+            choices.extend(section_choices[1])
+
+        kwargs["choices"] = choices
+
+        super().__init__(*args, **kwargs)
+        self.sections = sections
+        self.separator = separator
+
+    @property
+    def _display_choices(self) -> str:
+        lines = []
+
+        i = 1
+        for section in self.sections:
+            lines.append(f"{section[0]}")
```

Comment:
Could also do the `for section_name, choices in self.sections` , and then do a `lines.append(section_name)` here.